packages/exchange/src/exchange/langfuse_wrapper.py

```diff
@@ -14,29 +14,26 @@
 
 import os
 from typing import Callable
-from dotenv import load_dotenv
 from langfuse.decorators import langfuse_context
 import sys
 from io import StringIO
-from pathlib import Path
-from functools import wraps  # Add this import
+from functools import cache, wraps  # Add this import
 
-
-def find_package_root(start_path: Path, marker_file: str = "pyproject.toml") -> Path:
-    while start_path != start_path.parent:
-        if (start_path / marker_file).exists():
-            return start_path
-        start_path = start_path.parent
-    return None
+DEFAULT_LOCAL_LANGFUSE_HOST = "http://localhost:3000"
+DEFAULT_LOCAL_LANGFUSE_PUBLIC_KEY = "publickey-local"
+DEFAULT_LOCAL_LANGFUSE_SECRET_KEY = "secretkey-local"
 
 
+@cache
```

Comment:
nice! this is a lot cleaner

Reply:
how come we're memoizing this? seems like a premature optimization?

the env check is cheap enough it shouldn't matter. now `goose` can't get new env variables when set externally (i.e. next tracing event) because there is no cache invalidation

Reply:
Basically I did not change the existing behaviour. The purpose of the PR is mainly to remove the assumption of .env.langfuse.local file is part of exchange package, and rendering the local Langfuse credential from environment variables. (or it can be rendered from a environment file provided by goose application)

At the moment, I still leave the @cache method here as it is overkill to call the `auth_check` every time a function with observe_wrapper decorator is called.

By looking into the implementation, currently we rely on the `auth_check` to decide whether we want to trace the api call instead of deciding by whether the user passes `--tracing` option.  I think the improvement for this could be: 

- if goose session starts with `--tracing`, the application should raise with error message when error occurs with tracing
- If goose session starts without `--tracing`, then we don't need to perform `auth_check`